pkg/utils/iptables.go

```diff
@@ -0,0 +1,40 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/coreos/go-iptables/iptables"
+	"github.com/go-logr/logr"
+)
+
+// NewIPTables wraps the creation of IPTables to patch the path to the correct implementation binary.
+// It has been introduced to avoid the risk that the command doesn't work due to missing kernel modules.
+func NewIPTables(log logr.Logger, proto iptables.Protocol) (*iptables.IPTables, error) {
+	for _, suffix := range []string{"legacy", "nft"} {
+		if path := "/sbin/iptables-" + suffix; iptablesWorks(path) {
```

Comment:
Rather use exec.LookPath instead of hardcoding the path.

Reply:
I prefer to keep the hardcoded path, as it is well-defined by the alpine package.